### ISSUE TITLE
Version of the keyvals() getter which includes id/classes?

### ISSUE THREAD

**Issue by bpj** [nichtich/Pandoc-Elements#32]
The keyvals() getter is great, but I can easily imagine a scenario where I want the id and classes included.  What about a attr_as_hash() method? I volunteer to write it! ;-)

**Comment by nichtich** [comment]
It would still be a Hash::MultiValue but with (if given) attribute `id` and/or `class` (repeatable). How about accessor method `attr->all()`? The documentation would need to extended too, feel free to do so!

**Comment by bpj** [comment]
I'm thinking

```perl
    sub attr_as_hash {
        my $e = shift;
        if ( @_ ) {
            croak 'Method attr_as_hash() is not a setter';
        }
        my $h = $e->keyvals;
        $h->add( id => $e->id );
        $h->add( class => @{$e->classes} );
        return $h;
    }
```

**Comment by nichtich** [comment]
`id` an `class` should only be added if not empty, shouldn't they? I prefer the name `attr_hash` or `attributes` over `attr_as_hash`:

~~~perl
$element->attr; # full attributes, raw Pandoc format
$element->class; # scalar
$element->classes; # array
$element->keyvals; # multihash, key-value pairs only
$element->attributes; # full attributes, multihash
~~~

**Comment by bpj** [comment]
If I understand correctly id always exists. It may be an empty string
though. H::MV::set() won't add an empty entry if it is given only one
argument so there isn't really any need to double check. I wrote some tests
specifically to check that no `class => undef` entry is created, and that
the return value from `$hmv->as_hashref_mixed` is indeed identical to the
input to Pandoc::Elements::attributes() whether there were any classes in
the latter or not.

I don't think `attributes` would be a suitable name for this method. It's
too easily confused with Pandoc::Elements::attributes() and `$elem->attr`
which are rather different. I think `attr_hash` would probably be more
suitable. It describes what the method returns very accurately.

**Comment by nichtich** [comment]
ok, go for attr_hash :-)

**Comment by nichtich** [comment]
I spent some time to merge #33 after realizing that `$e->keyvals` already respects `id` and `class` as setter, so it should also include them in the return value. This is a much cleaner solution than adding another method, so I implemented it in release 0.21.
